tests/integration/interaction/test_responsive.py

```diff
@@ -156,3 +162,132 @@ def test_responsive_plot_starts_at_correct_size(output_file_url, selenium):
     # Canvas width should be just under 1000
     assert canvas.size['width'] > 900
     assert canvas.size['width'] < 1000
+
+
+def test_responsive_resizes_width_and_height(output_file_url, selenium):
+    # Test that a Bokeh plot embedded in a desktop-ish setting (e.g.
+    # a Phosphor widget) behaves well w.r.t. resizing.
+    
+    # We want the aspect ratio of the initial plot to be maintained, but we
+    # can't measure it perfectly so we test against bounds.
+    aspect_ratio = 2
+    plot_height = 400
+    plot_width = 400 * aspect_ratio
+    lower_bound = aspect_ratio * 0.95
+    upper_bound = aspect_ratio * 1.05
+
+    # Define our html template for out plots
+    template = Template('''<!DOCTYPE html>
+    <html lang="en">
+        <head>
+            <meta charset="utf-8">
+            {{ js_resources }}
+            {{ css_resources }}
+        </head>
+        <body>
+        {{ plot_div.red }}
+        {{ plot_script }}
+        <script>
+        // Set things up to resize the plot on a window resize. You can play with
+        // the arguments of resize_width_height() to change the plot's behavior.
+        var plot_resize_setup = function () {
+            var plotid = Object.keys(Bokeh.index)[0]; // assume we have just one plot
+            var plot = Bokeh.index[plotid];
+            var plotresizer = function() {
+                // arguments: use width, use height, maintain aspect ratio
+                plot.resize_width_height(true, true, true);
+            };
+            window.addEventListener('resize', plotresizer);
+            plotresizer();
+        };
+        window.addEventListener('load', plot_resize_setup);
+        </script>
+        <style>
+        /* Need this to get the page in "desktop mode"; not having an infinite height.*/
+        html, body {height: 100%;}
+        </style>
+        </body>
+    </html>
+    ''')
+    
+    PLOT_OPTIONS = dict(plot_width=plot_width, plot_height=plot_height)
+    SCATTER_OPTIONS = dict(size=12, alpha=0.5)
+    
+    data = lambda: [random.choice([i for i in range(100)]) for r in range(10)]
+    
+    red = figure(responsive=False, tools='pan', **PLOT_OPTIONS)
+    red.scatter(data(), data(), color="red", **SCATTER_OPTIONS)
+    
+    resources = INLINE
+    
+    js_resources = resources.render_js()
+    css_resources = resources.render_css()
+    
+    script, div = components({'red': red})
+    
+    html = template.render(js_resources=js_resources,
+                        css_resources=css_resources,
+                        plot_script=script,
+                        plot_div=div)
+    
+    filename = output_file_url.split('/', 3)[-1]  # strip "http://xxxx:yyyy/"
```

Comment:
It would be easy to expose the `file_path` as a fixture too so that you can just use `file_path` like `output_file_url` and not have to manually produce it.

If this PR is urgent, then we can just change it later. If not I could push the change to the branch.